### PR TITLE
Introduce configurable PartitionDistributor component

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

This PR extracts a new interface from the `PartitionDistributor`, renaming the previous implementation to `RoundRobinPartitionDistributor`, and allows configuring the `RaftPartitionGroup` with an arbitrary implementation of this interface (which by default is the `RoundRobinPartitionDistributor`). This is in preparation for #6266, which will introduce a new configurable partitioning scheme.

## Related issues

related to #6266 in preparation for the new partitioning scheme

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: atomix/cluster/src/main/java/io/atomix/raft/partition/PartitionDistributor.java [hunk 1]
```diff
@@ -15,84 +15,36 @@
...
+   * Generates the partition distribution based on the given list of partition IDs, cluster members,
+   * and the replication factor. The set of partitions returned is guaranteed to be correctly
+   * replicated.
+   *
+   * @param clusterMembers the set of members that can own partitions
+   * @param sortedPartitionIds a sorted list of partition IDs
+   * @param replicationFactor the replication factor for each partition
+   * @return a set of distributed partitions, each specifying which members they belong to
+   */
+  Set<PartitionMetadata> generatePartitionDistribution(
```
Comment (deepthidevaki) [inline atomix/cluster/src/main/java/io/atomix/raft/partition/PartitionDistributor.java:48]: :thought_balloon: Is the name "generate" suitable when we have fixed distribution?

**Review by deepthidevaki — APPROVED** [review]
Thanks. Just have a minor comment. No blockers.

File: atomix/cluster/src/main/java/io/atomix/raft/partition/PartitionDistributor.java (hunk 1 above)
Comment (npepinpe) [reply atomix/cluster/src/main/java/io/atomix/raft/partition/PartitionDistributor.java:48]: Honestly I didn't even think about it, I just kept the name :smile: What about `distributePartitions`?

Comment (npepinpe) [reply atomix/cluster/src/main/java/io/atomix/raft/partition/PartitionDistributor.java]: I've renamed it `distributePartitions` instead

**Comment by npepinpe** [comment]
bors merge

**Pull request merged**